### PR TITLE
feat(DataTable): Add sub-rows handling based on records structure

### PR DESCRIPTION
**Pull request by cbaltatescu**
These changes add the ability to render a hierarchical records structure, however the DataTable component is not responsible per se for calculating th ancestors and descendents, it just cares about the order of the records and the _level_ attribute (level N is a parent of level N-1).
These are calculated and provided from the component that renders the DataTable.
The same component will be responsible for sorting in this case.
Basically the hierarchy is is just **visual**, the records need to be provided arranged so that they appear to be related (child follows parent, level flag translates to indenting, represented as CSS padding-left).

### CODE REVIEW

File: src/components/DataTable.jsx [hunk 1]
```diff
@@ -134,35 +142,39 @@ export default class DataTable extends Base {
...
     });
     return filteredRecord;
-  }
+  };
 
-  filteredSubRecords = (fields) => (record) =>
-    fields.some((val) =>
-      this.props.filterRecords.some((filterVal) =>
-        (filterVal[val] === record[val])));
+  filteredSubRecords = fields => record => fields.some(val => this.props.filterRecords.some(filterVal => filterVal[val] === record[val]));
```
Comment (ryanballa) [inline src/components/DataTable.jsx]: This seems more difficult to read than how it was before
